compiler/src/iree/compiler/Codegen/Common/EncodingInfo.h

```diff
@@ -0,0 +1,70 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_COMPILER_SRC_IREE_COMPILER_CODEGEN_COMMON_ENCODINGINFO_H_
+#define IREE_COMPILER_SRC_IREE_COMPILER_CODEGEN_COMMON_ENCODINGINFO_H_
+
+#include "iree-dialects/Dialect/LinalgExt/Passes/Passes.h"
+#include "iree/compiler/Dialect/HAL/IR/HALTypes.h"
+
+namespace mlir {
+namespace iree_compiler {
+
+enum class MatmulType {
+  F32F32F32,
+  I8I8I32,
+};
+
+enum class MatmulOperandRole {
+  LHS,
+  RHS,
+  RHS_TRANSPOSE,
+  RESULT,
+};
+
+struct MatmulTileParams {
+  int64_t M = 1;
+  int64_t K = 1;
+  int64_t N = 1;
+};
+
+/// Extracts encoding from the `tensorType` if specified.
+Optional<IREE::LinalgExt::TensorEncoding> getEncoding(
+    RankedTensorType tensorType);
+
+Optional<MatmulType> getMatmulType(IREE::LinalgExt::TensorEncoding encoding);
+
+Optional<MatmulOperandRole> getMatmulOperandRole(
+    IREE::LinalgExt::TensorEncoding encoding);
+
+void adjustTileSizesToNarrowStaticShape(
+    IREE::LinalgExt::MaterializeEncodingInfo &encodingInfo,
+    ArrayRef<int64_t> shape);
+
+IREE::LinalgExt::MaterializeEncodingInfo chooseEncodingInfoForMatmul(
+    MatmulType type, MatmulOperandRole operandRole,
+    MatmulTileParams tileParams);
+
+IREE::LinalgExt::MaterializeEncodingValueFn getMaterializeEncodingValueFn(
+    IREE::HAL::ExecutableTargetAttr targetAttr);
+
+void populateMaterializeEncodingIntoPackUnPackPatterns(
+    RewritePatternSet &patterns,
+    IREE::LinalgExt::MaterializeEncodingConversionTarget &target,
+    IREE::LinalgExt::MaterializeEncodingTypeConverter &typeConverter,
+    IREE::LinalgExt::MaterializeEncodingValueFn materializeEncodingValueFn);
+
+// TODO(hanchung): Move the method to VMVX/EncodingInfo.h. This is required by
```

Comment:
You can add me to this TODO to break this dependency.

Reply:
Incidentally, #11818 makes TileAndDistributeToWorkgroupPass no longer use this  (and no longer use getMaterializeEncodingValueFn).

Reply:
If @bjacob's PR works, that would be great. We can break the dep between Common/ and VMVX/!